src/main/java/com/onlinebookstore/controller/BookController.java

```diff
@@ -36,6 +37,7 @@ public class BookController {
 
     @Operation(summary = "Get all books", description = "Get a list of all available books")
     @ApiResponse(responseCode = "200", description = "Operation successful")
+    @PreAuthorize("hasRole('USER')")
```

Comment:
what if admin also want to get all books? 
Preauthorize not needed here it should be allowed for all authentiphicated users same as getBookById and search

Reply:
Ok, maybe the task description is a bit unclear. That's why I also added 'user' role to admin user roles list. Deleting...